merkle/src/store.rs

```diff
@@ -1,22 +1,74 @@
 use failure::Error;
-use merkle::Element;
+use merkle::{Element, next_pow2, log2_pow2};
 use positioned_io::{ReadAt, WriteAt};
-use std::fs::File;
+use std::fs::{File, OpenOptions};
+use std::path::Path;
 use std::marker::PhantomData;
 use std::ops::{self, Index};
 use tempfile::tempfile;
 
 pub type Result<T> = std::result::Result<T, Error>;
 
+
+#[derive(Debug)]
+pub struct StoreConfig {
+    /// A directory in which data (a merkle tree) can be persisted.
+    pub path: String,
+
+    /// A unique identifier used to help specify the on-disk store
+    /// location for this particular data.
+    pub id: String,
+
+    /// The number of merkle tree levels to cache in memory, starting
+    /// at the root.  A value larger than the number of levels in the
+    /// tree will cache the entire merkle tree.
+    pub levels: usize,
+}
+
+impl StoreConfig {
+    pub fn new(path: String, id: String, levels: usize) -> Result<StoreConfig> {
+        Ok(StoreConfig {
+            path,
+            id,
+            levels
+        })
+    }
+
+    // Deterministically create the merkle_tree_path on-disk location
+    // from a path and specified id.
+    pub fn merkle_tree_path(path: &str, id: &str) -> String {
```

Comment:
- This should return a `PathBuf`
- maybe not call it merkle_tree_path, given that we use it generally for disk stores